drivers/ethernet/eth_native_posix.c

```diff
@@ -0,0 +1,268 @@
+/*
+ * Copyright (c) 2018 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ *
+ * Ethernet driver for native posix board. This is meant for network
+ * connectivity between host and Zephyr.
+ */
+
+#define SYS_LOG_DOMAIN "eth-posix"
+#define SYS_LOG_LEVEL CONFIG_SYS_LOG_ETH_NATIVE_POSIX_LEVEL
+
+#include <logging/sys_log.h>
+#include <stdio.h>
+
+#include <kernel.h>
+
+#include <stdbool.h>
+#include <errno.h>
+#include <stddef.h>
+
+#include <net/net_pkt.h>
+#include <net/net_core.h>
+#include <net/net_if.h>
+#include <net/ethernet.h>
+
+#include "eth_native_posix_priv.h"
+
+#if defined(CONFIG_NET_L2_ETHERNET)
+#define _ETH_L2_LAYER ETHERNET_L2
+#define _ETH_L2_CTX_TYPE NET_L2_GET_CTX_TYPE(ETHERNET_L2)
+#define _ETH_MTU 1500
+#endif
+
+#define NET_BUF_TIMEOUT MSEC(10)
+
+struct eth_context {
+	u8_t recv[_ETH_MTU + sizeof(struct net_eth_hdr)];
+	u8_t send[_ETH_MTU + sizeof(struct net_eth_hdr)];
+	u8_t mac_addr[6];
+	struct net_linkaddr ll_addr;
+	struct net_if *iface;
+	const char *if_name;
+	int dev_fd;
+	bool init_done;
+	bool status;
+};
+
+NET_STACK_DEFINE(RX_ZETH, eth_rx_stack, CONFIG_ETH_NATIVE_POSIX_RX_STACK_SIZE,
+		 CONFIG_ETH_NATIVE_POSIX_RX_STACK_SIZE);
+static struct k_thread rx_thread_data;
+
+/* TODO: support multiple interfaces */
+static struct eth_context eth_context_data;
+
+static struct eth_context *get_context(struct net_if *iface)
+{
+	return net_if_get_device(iface)->driver_data;
+}
+
+static int eth_send(struct net_if *iface, struct net_pkt *pkt)
+{
+	struct eth_context *ctx = get_context(iface);
+	struct net_buf *frag;
+	int count = 0;
+
+	/* First fragment contains link layer (Ethernet)
+	 * headers so send it first.
+	 */
+	count = net_pkt_ll_reserve(pkt) + pkt->frags->len;
+	memcpy(ctx->send, net_pkt_ll(pkt), count);
+
+	/* Then the remaining data */
+	frag = pkt->frags->frags;
+	while (frag) {
+		memcpy(ctx->send + count, frag->data, frag->len);
+		count += frag->len;
+		frag = frag->frags;
+	}
+
+	net_pkt_unref(pkt);
+
+	eth_write_data(ctx->dev_fd, ctx->send, count);
+
+	return 0;
+}
+
+#if !defined(CONFIG_ETH_NATIVE_POSIX_RANDOM_MAC)
+static int mac_addr_from_str(struct eth_context *ctx, const char *src)
```

Comment:
I think it will be wise to put that function somewhere in a unique place, slip uses it also.

Reply:
Perhaps, I will investigate this more.